internal/updater/utils/html/parsing.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/andybalholm/cascadia"
+	"golang.org/x/net/html"
+)
+
+var (
+	ErrAttrNotFound = errors.New("matching attribute not found")
+)
+
+func GetText(n *html.Node) string {
+	return n.FirstChild.Data
+}
+
+func GetAttr(n *html.Node, key string) (string, error) {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return attr.Val, nil
+		}
+	}
+
+	return "", ErrAttrNotFound
+}
+
+func CheckID(n *html.Node, id string) bool {
+	if n.Type == html.ElementNode {
+		s, err := GetAttr(n, "id")
+
+		if err == nil && s == id {
+			return true
+		}
+	}
+
+	return false
+}
+
+func Traverse(n *html.Node, id string) *html.Node {
+	if CheckID(n, id) {
+		return n
+	}
+
+	for c := n.FirstChild; c != nil; c = c.NextSibling {
+		res := Traverse(c, id)
+
+		if res != nil {
+			return res
+		}
+	}
+
+	return nil
+}
+
+func GetNodeByID(n *html.Node, id string) *html.Node {
+	return Traverse(n, id)
+}
+
```

Comment:
This looks rather unneeded, one might just use `Traverse` directly

```suggestion
```

Reply:
Renamed Traverse to GetNodeByID, and removed this function entirely